src/react.ts

```diff
@@ -92,26 +61,12 @@ type Create = {
   <T>(): <Mos extends [StoreMutatorIdentifier, unknown][] = []>(
     initializer: StateCreator<T, [], Mos>,
   ) => UseBoundStore<Mutate<StoreApi<T>, Mos>>
-  /**
-   * @deprecated Use `useStore` hook to bind store
-   */
-  <S extends StoreApi<unknown>>(store: S): UseBoundStore<S>
 }
 
 const createImpl = <T>(createState: StateCreator<T, [], []>) => {
-  if (
-    import.meta.env?.MODE !== 'production' &&
-    typeof createState !== 'function'
-  ) {
-    console.warn(
-      "[DEPRECATED] Passing a vanilla store will be unsupported in a future version. Instead use `import { useStore } from 'zustand'`.",
-    )
-  }
-  const api =
-    typeof createState === 'function' ? createStore(createState) : createState
+  const api = createStore(createState)
 
-  const useBoundStore: any = (selector?: any, equalityFn?: any) =>
-    useStore(api, selector, equalityFn)
+  const useBoundStore: any = (selector?: any) => useStore(api, selector)
```

Comment:
Aside: I'm wondering if we can review these `any` types and potentially replace them with something more specific.

Reply:
Definitely, anything other than `any` is always better than `any`